SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -200,31 +203,73 @@ private Task ProcessPendingErrorsAsync()
 
         private Task SendCrashReportsOrAwaitUserConfirmationAsync()
         {
-            return HandleUserConfirmationAsync();
+            bool alwaysSend = ApplicationSettings.GetValue(PrefKeyAlwaysSend, false);
+
+            return Task.Run(async () =>
+            {
+                if (_unprocessedManagedErrorLogs.Count() > 0)
+                {
+                    // Check for always send: this bypasses user confirmation callback.
+                    if (alwaysSend)
+                    {
+                        AppCenterLog.Debug(LogTag, "The flag for user confirmation is set to AlwaysSend, will send logs.");
+                        await HandleUserConfirmationAsync(UserConfirmation.Send);
+                        return;
+                    }
+
+                    if (ShouldAwaitUserConfirmation?.Invoke() ?? false)
```

Comment:
While it is correct, I found this condition a little confusing. I think it is because it's doing a lot of things at once: making sure that `ShouldAwaitUserConfirmation` is not null, invoking it, and coalescing it to a default value. Up to you if you want to change it, but just letting you know it took me a second to follow this.

Reply:
Agreed with @achocron, the readability can be improved.